code/src/components/TimeData.js

```diff
@@ -0,0 +1,64 @@
+import React from "react"
+import styled from 'styled-components'
+import Moment from 'react-moment'
+import moment from "moment"
+
+const TimeStamp = styled.p`
+  font-size: 12px;
+  margin: 2px;
+`
+
+export const TimeData = ({ ...item }) => {
+
+    const checkTimeLeft = (item) => {
+        if (!item.dueDate) {
+            return
+        } else if (item.dueDate) {
+            const due = moment(item.dueDate);
+            const timeDiff = moment(due.diff(Date.now()))
+            const duration = moment.duration(timeDiff._i)
+            if (duration._data.days >= 7 || duration._data.years) {
+                return `Due: ${item.dueDate}`
+            } else if (duration._data.days < 0 || duration._data.hours < 0 || duration._data.minutes < 0) {
+                if (duration._data.days) {
+                    return `Due ${duration._data.days.toString().substr(1)} days ago`
+                } else {
+                    return `Due ${duration._data.hours.toString().substr(1)} hours ago`
+                }
+            } else if (duration._data.days < 7 && duration._data.days !== 0) {
+                return `Due in ${duration._data.days} days`
+            } else if (duration._data.days === 0) {
+                return `Due in ${duration._data.hours} hours`
+            } else if (duration._data.days === 0 && duration._data.hours === 0) {
+                return `Due in ${duration._data.minutes}} minutes`
+            }
+        }
```

Comment:
I haven't really gone through all of this code but it seems like something here is a bit off?

[Screen Recording 2020-04-30 at 16.29.18.mov.zip](https://github.com/Technigo/project-todos/files/4558926/Screen.Recording.2020-04-30.at.16.29.18.mov.zip)

Reply:
Thanks for such in-depth feedback Jenny! I really don't understand what this problem with the due date is though, as it works perfectly on my computer (mac running chrome) and on my iPhone running safari...🤔

Reply:
I'll investigate a bit in this, I'll get back to you!